addons/web/static/src/js/core/custom_hooks.js

```diff
@@ -36,8 +36,51 @@ odoo.define('web.custom_hooks', function () {
             component.__willFocus = selector;
         };
     }
+    /**
+     * When a component needs to set up generic event handling on its root node
+     * and makes it available for all subclasses
+     * Useful in many parts of Odoo, when a subclass of a component
+     * usually *adds* behavior relatively to its parent class
+     *
+     * Example: navigation event handling in Fields
+     *
+     * Usage:
+     *   In the constructor of the OWL component
+     *
+     * Note:
+     *   When selector contains a native querySelector, event delegation is activated
+     *   Components that alter Event.target (e.g. Portal) are not expected
+     *   to behave as expected in that setting
+     *
+     * @param {String} selector. A string containing the event name, a space
+     *     and then a valid native JS selector
+     * @param {Function}. The handler of the event; will be bound to component
+     */
+    function useListener(selector, handler) {
+        const comp = Component.current;
+        if (!(typeof handler === 'function')) throw new Error('The handler must be a function');
+        const match = /^(\S+)(\s+(.*))?$/.exec(selector);
+        const evName = match[1];
+        const querySelector = match[2];
+
+        let boundHandler = handler.bind(comp);
+        if (querySelector) {
+            boundHandler = function (ev) {
+                if (ev.target && ev.target.matches(querySelector)) {
```

Comment:
i think you must iterate over each node from the target to the root node, following the parent relation, to find one matching the selector